packages/tailwindcss-language-service/src/util/css-vars.test.ts

```diff
@@ -1,26 +1,53 @@
 import { expect, test } from 'vitest'
 import { replaceCssVarsWithFallbacks } from './css-vars'
+import { State } from './state'
+import { DesignSystem } from './v4'
 
 test('replacing CSS variables with their fallbacks (when they have them)', () => {
-  expect(replaceCssVarsWithFallbacks('var(--foo, red)')).toBe(' red')
-  expect(replaceCssVarsWithFallbacks('var(--foo, )')).toBe(' ')
+  let map = new Map<string, string>([
+      ['--foo', 'blue'],
+  ])
 
-  expect(replaceCssVarsWithFallbacks('rgb(var(--foo, 255 0 0))')).toBe('rgb( 255 0 0)')
-  expect(replaceCssVarsWithFallbacks('rgb(var(--foo, var(--bar)))')).toBe('rgb( var(--bar))')
+  let state: State = {
+    enabled: true,
+    designSystem: {
+      resolveThemeValue: (name) => map.get(name) ?? null,
+    } as DesignSystem,
+  }
+
+  expect(replaceCssVarsWithFallbacks(state, 'var(--foo, red)')).toBe(' red')
+  expect(replaceCssVarsWithFallbacks(state, 'var(--foo, )')).toBe(' ')
+
+  expect(replaceCssVarsWithFallbacks(state, 'rgb(var(--foo, 255 0 0))')).toBe('rgb( 255 0 0)')
+  expect(replaceCssVarsWithFallbacks(state, 'rgb(var(--foo, var(--bar)))')).toBe('rgb( var(--bar))')
 
   expect(
-    replaceCssVarsWithFallbacks('rgb(var(var(--bar, var(--baz), var(--qux), var(--thing))))'),
+    replaceCssVarsWithFallbacks(
+      state,
+      'rgb(var(var(--bar, var(--baz), var(--qux), var(--thing))))',
+    ),
   ).toBe('rgb(var(var(--bar, var(--baz), var(--qux), var(--thing))))')
 
   expect(
     replaceCssVarsWithFallbacks(
+      state,
       'rgb(var(--one, var(--bar, var(--baz), var(--qux), var(--thing))))',
     ),
   ).toBe('rgb(  var(--baz), var(--qux), var(--thing))')
 
   expect(
     replaceCssVarsWithFallbacks(
+      state,
       'color-mix(in srgb, var(--color-primary, oklch(0 0 0 / 2.5)), var(--color-secondary, oklch(0 0 0 / 2.5)), 50%)',
     ),
   ).toBe('color-mix(in srgb,  oklch(0 0 0 / 2.5),  oklch(0 0 0 / 2.5), 50%)')
+
+  // Fallbacks take precedence over theme values
+  expect(replaceCssVarsWithFallbacks(state, 'var(--foo, red)')).toBe(' red')
```

Comment:
I understand that this is unlikely to be generated by TW's core utilities but I’m not sure I understand the concept behind it even. Shouldn't the theme var always get precedence just like it would in the browser? 🤔

Reply:
Oh yeah that's a good point. I was basically making a change that lines up with how intellisense _reads things today_ but you're right the value from the theme taking precedence makes sense. I'll swap the behavior here.